src/datatype/memory_veci8.rs

```diff
@@ -0,0 +1,241 @@
+use crate::prelude::*;
+use pgrx::pg_sys::Datum;
+use pgrx::pg_sys::Oid;
+use pgrx::pgrx_sql_entity_graph::metadata::ArgumentError;
+use pgrx::pgrx_sql_entity_graph::metadata::Returns;
+use pgrx::pgrx_sql_entity_graph::metadata::ReturnsError;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlMapping;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlTranslatable;
+use pgrx::FromDatum;
+use pgrx::IntoDatum;
+use std::alloc::Layout;
+use std::cmp::Ordering;
+use std::ops::Deref;
+use std::ops::DerefMut;
+use std::ptr::NonNull;
+
+pub const VECI8_KIND: u8 = 3;
+
+/// Veci8 utilizes int8 for data storage, originally derived from Vecf32.
+/// Given a vector of F32, [a_0, a_1, a_2, ..., a_n], we aim to find the maximum and minimum values. The maximum value, max, is the greatest among {a_0, a_1, a_2, ..., a_n}, and the minimum value, min, is the smallest.
+/// We can transform F32 to I8 using the formula (a - (max + min) / 2) / (max - min) * 255, resulting in a vector of I8, [b_0, b_1, b_2, ..., b_n]. Here 255 is the range size that the int8 type can cover, which is the difference between -128 and 127.
+/// Converting I8 back to F32 can be achieved by using the formula b * (max - min) / 255 + (max + min) / 2, which gives us a vector of F32, albeit with a slight loss of precision.
+/// We use alpha to represent (max - min) / 255, and offset to represent (max + min) / 2 here.
+#[repr(C, align(8))]
+pub struct Veci8Header {
+    varlena: u32,
+    len: u16,
+    kind: u8,
+    reserved: u8,
+    alpha: F32,
+    offset: F32,
+    phantom: [I8; 0],
```

Comment:
Let's store `sum` and `l2_norm` here too.

Reply:
I've thought about it.  I think the precomputed results are only useful for indexes, maybe it isn't necessary to store it in postgres？

Reply:
Just keep in the struct for simplicity. One extra fp32 is not a big deal

Reply:
get it.